src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -813,7 +815,7 @@ void SpatialPooler::updatePermanencesForColumn_(vector<Real>& perm,
   numConnected = 0;
   for (UInt i = 0; i < perm.size(); ++i)
   {
-    if (perm[i] >= synPermConnected_)
+    if (perm[i] >= synPermConnected_ - PERMANENCE_EPSILON)
```

Comment:
I don't understand this. If the two values are exactly equal (use of `>=` instead of `>` makes me think that is significant) then the epsilon will result in the wrong outcome. If there is some case where different platforms have a slight difference then I don't even see how this would help.